### PR TITLE
Add document for some porous flow tests.

### PR DESCRIPTION
**Pull request by ChiCoTran**
Refs #26240

### CODE REVIEW

File: modules/porous_flow/doc/content/modules/porous_flow/tests/capillary_pressure/capillary_pressure_tests.md [hunk 1]
```diff
@@ -1,3 +1,35 @@
 # Capillary pressure test descriptions
 
-TODO
+#### The detail documentation of the implementation of this code packet can be founded in [here](porous_flow/capillary_pressure.md).
```
Comment (cpgr) [inline modules/porous_flow/doc/content/modules/porous_flow/tests/capillary_pressure/capillary_pressure_tests.md:3]:
```suggestion
The details of the capillary pressure curve implementations can be found in [here](porous_flow/capillary_pressure.md).
```

File: modules/porous_flow/doc/content/modules/porous_flow/tests/capillary_pressure/capillary_pressure_tests.md [hunk 2]
```diff
@@ -1,3 +1,35 @@
...
+
+## Brooks-Corey
+
+!media capillary_pressure/brook_corey.png style=width:100%;margin-left:10px; caption=Brooks-Corey Capillary Pressure test cases id=brookcoreyPc
+
```
Comment (cpgr) [inline modules/porous_flow/doc/content/modules/porous_flow/tests/capillary_pressure/capillary_pressure_tests.md:8]: I would expect a better match here - can you double check the figure?

Comment (cpgr) [reply modules/porous_flow/doc/content/modules/porous_flow/tests/capillary_pressure/capillary_pressure_tests.md:8]: Also a typo in the heading of this figure

File: modules/porous_flow/doc/content/modules/porous_flow/tests/capillary_pressure/capillary_pressure_tests.md [hunk 3]
```diff
@@ -1,3 +1,35 @@
...
+The input file for log-extension off test:
```
Comment (cpgr) [inline modules/porous_flow/doc/content/modules/porous_flow/tests/capillary_pressure/capillary_pressure_tests.md:9]: Can you reword this to say that there are two test cases - with and without the log extension at low saturation's

File: modules/porous_flow/doc/content/modules/porous_flow/tests/capillary_pressure/capillary_pressure_tests.md [hunk 4]
```diff
@@ -1,3 +1,35 @@
...
+!listing modules/porous_flow/test/tests/capillary_pressure/brooks_corey2.i
+
+
+
+## van Genuchten
+
+
+
+!media capillary_pressure/vangen.png style=width:100%;margin-left:10px; caption=van Genuchten Capillary Pressure test cases id=vangenPc
+
```
Comment (cpgr) [inline modules/porous_flow/doc/content/modules/porous_flow/tests/capillary_pressure/capillary_pressure_tests.md:24]: Maybe extend the x range to include 0 to show what happens? If the y axis gets too large, you could use a log scaling on that axis, or just limit the yrange

Comment (cpgr) [reply modules/porous_flow/doc/content/modules/porous_flow/tests/capillary_pressure/capillary_pressure_tests.md:24]: Also a typo in the plot heading - can you carefully check these?

File: modules/porous_flow/doc/content/modules/porous_flow/tests/capillary_pressure/capillary_pressure_tests.md [hunk 5]
```diff
@@ -1,3 +1,35 @@
...
+The input file for log-extension off test:
```
Comment (cpgr) [inline modules/porous_flow/doc/content/modules/porous_flow/tests/capillary_pressure/capillary_pressure_tests.md:25]: Similar to above, mention that there are three test cases here:

File: modules/porous_flow/doc/content/modules/porous_flow/tests/relperm/relperm_tests.md [hunk 6]
```diff
@@ -1,3 +1,83 @@
 # Relative permeability test descriptions
 
-TODO
+#### The detail documentation of the implementation of this code packet can be founded in [here](porous_flow/relative_permeability.md).
```
Comment (cpgr) [inline modules/porous_flow/doc/content/modules/porous_flow/tests/relperm/relperm_tests.md:3]:
```suggestion
The detail documentation of the implementation of this code packet can be founded in [here](porous_flow/relative_permeability.md).
```

Change this to be similar to the capillary pressure tests

File: modules/porous_flow/doc/content/modules/porous_flow/tests/relperm/relperm_tests.md [hunk 7]
```diff
@@ -1,3 +1,83 @@
...
+!media relperm/brook_corey1.png style=width:100%;margin-left:10px; caption=Brooks-Corey relative permeability Test case 1 id=brookcorey1
+
+#### Test 2
+
+The input file for this test:
+
+!listing modules/porous_flow/test/tests/relperm/brooks_corey2.i
+
+!media relperm/brook_corey2.png style=width:100%;margin-left:10px; caption=Brooks-Corey relative permeability Test case 2 id=brookcorey2
+
```
Comment (cpgr) [inline modules/porous_flow/doc/content/modules/porous_flow/tests/relperm/relperm_tests.md:23]: can you use MOOSE in the legend for this (and all) figures rather than moose

File: modules/porous_flow/doc/content/modules/porous_flow/tests/relperm/relperm_tests.md [hunk 8]
```diff
@@ -1,3 +1,83 @@
...
+!media relperm/corey3.png style=width:100%;margin-left:10px; caption=Corey relative permeability Test case 3 id=corey3
+
+#### Test 4
+
+The input file for this test:
+
+!listing modules/porous_flow/test/tests/relperm/corey4.i
+
+!media relperm/corey4.png style=width:100%;margin-left:10px; caption=Corey relative permeability Test case 4 id=corey4
+
```
Comment (cpgr) [inline modules/porous_flow/doc/content/modules/porous_flow/tests/relperm/relperm_tests.md:65]: Does kr1 go to 10 here? can you show the complete y range.

File: modules/porous_flow/doc/content/modules/porous_flow/tests/relperm/relperm_tests.md [hunk 9]
```diff
@@ -1,3 +1,83 @@
...
+!media relperm/vangen1.png style=width:100%;margin-left:10px; caption=van Genuchten relative permeability Test case 1 id=vangen1
+
+#### Test 2
+
+The input file for this test:
+
+!listing modules/porous_flow/test/tests/relperm/vangenuchten2.i
+
+!media relperm/vangen2.png style=width:100%;margin-left:10px; caption=van Genuchten relative permeability Test case 2 id=vangen2
```
Comment (cpgr) [inline modules/porous_flow/doc/content/modules/porous_flow/tests/relperm/relperm_tests.md:83]: Maybe add the references for all of the various formulations, rather than just the Corey one?

File: modules/porous_flow/doc/content/modules/porous_flow/tests.md [hunk 10]
```diff
@@ -28,7 +28,6 @@ Many of the PorousFlow tests were created before MOOSE's current documentation s
 - [actions](tests/actions/actions_tests.md)
 - [fluids](tests/fluids/fluids_tests.md)
 - [numerical_diffusion](tests/numerical_diffusion/numerical_diffusion_tests.md)
-- [adv_diff_bcs](tests/adv_diff_bcs/adv_diff_bcs_tests.md)
 - [aux_kernels](tests/aux_kernels/aux_kernels_tests.md)
 - [flux_limited_TVD_advection](tests/flux_limited_TVD_advection/flux_limited_TVD_advection_tests.md)
 - [basic_advection](tests/basic_advection/basic_advection_tests.md)
```
Comment (cpgr) [inline modules/porous_flow/doc/content/modules/porous_flow/tests.md:33]: Can you move the tests that you documented to the top list (so out of the list of undocumented tests)? Also, change [relperm] to [Relative permeability], etc

**Comment by cpgr** [comment]
Good work @ChiCoTran

File: modules/porous_flow/doc/content/modules/porous_flow/tests/capillary_pressure/capillary_pressure_tests.md (hunk 4 above)
Comment (ChiCoTran) [reply modules/porous_flow/doc/content/modules/porous_flow/tests/capillary_pressure/capillary_pressure_tests.md:28]: Can you have a look on this figure again for the analytical curve to see if it is correct?

**Comment by cpgr** [comment]
This one's nearly ready - just need to check the fit for a couple of plots which don't match exactly. I'll take a look

**Comment by cpgr** [comment]
One last thing before we merge this - can you move the Rehbinder test link from the bottom undocumented list to the top list in the test index page

**Comment by ChiCoTran** [comment]
> One last thing before we merge this - can you move the Rehbinder test link from the bottom undocumented list to the top list in the test index page

I have fixed that.

**Review by cpgr — APPROVED** [review]
Good work again!
